Adding in generic update instructions and fixing some links. (#99)
timescale/docs: timescaledb/how-to-guides/update-timescaledb/index.md

 --------------------|-------------------------------
 1.7                 | 9.6, 10, 11, 12
 2.0                 | 11, 12
 2.1-2.3             | 11, 12, 13
 2.4+                | 12, 13

> <highlight type="tip">
> If you need to upgrade PostgreSQL first, please see [our documentation](/timescaledb/latest/how-to-guides/update-timescaledb/upgrade-postgresql/).
> 
> **Please note** that we always recommend that PostgreSQL and TimescaleDB be updated
> as separate actions to ensure each process completes properly. For example, if you 
> are currently running PostgreSQL 10 and TimescaleDB 1.7.5, the recommended upgrade 
> path to PostgreSQL 13 and TimescaleDB 2.2 would be:
> 
> 1. Upgrade PostgreSQL 10 to PostgreSQL 12
> 1. Update TimescaleDB 1.7.5 to TimescaleDB 2.2 on PostgreSQL 12
> 1. Upgrade PostgreSQL 12 to PostgreSQL 13 with TimescaleDB 2.2 installed
> 
> </highlight>

Reviewer: ```suggestion
If you need to upgrade PostgreSQL first, 
see [our documentation](/timescaledb/latest/how-to-guides/update-timescaledb/upgrade-postgresql/).

<highlight type="tip">
We always recommend that you update PostgreSQL and TimescaleDB as 
separate actions to make sure that each process completes properly. 
For example, if you are currently running PostgreSQL 10 and 
TimescaleDB 1.7.5, and you want to upgrade to PostgreSQL 13 and
TimescaleDB 2.2, upgrade in this order:
1. Upgrade PostgreSQL 10 to PostgreSQL 12
1. Update TimescaleDB 1.7.5 to TimescaleDB 2.2 on PostgreSQL 12
1. Upgrade PostgreSQL 12 to PostgreSQL 13 with TimescaleDB 2.2 installed
</highlight>
```

Is there a particular reason why you use 'update' for TimescaleDB and 'upgrade' for pg?

Author: Mostly semantics. The Postgres documentation actually talks about "upgrading" Postgres. With extensions, the command is literally:

```ALTER EXTENION timescaledb UPDATE;```

My impression is that in the past we've felt that it was worth distinguishing. It does get (slightly) confusing trying to maintain it when writing, however.  ;-)

Reviewer: You're breaking my heart here 😂
